PyU4V/tools/openstack/migrate.py

```diff
@@ -0,0 +1,149 @@
+        sys.exit()
+
+for masking_view in masking_view_list:
+    if utils.validate_masking_view(masking_view, is_revert):
+        txt = 'Do you want to migrate %s. Y/N or X(exit): ' % masking_view
+        txt_out = input(txt)
+        if utils.check_input(txt_out, 'Y'):
+            if is_revert:
+                if utils.check_mv_for_migration(masking_view, is_revert):
+                    no_action = False
+                    utils.smart_print(
+                        'NEW MASKING VIEW IS %s',
+                        migrate_utils.DEBUG, masking_view)
+                    mv_components = (
+                        utils.get_elements_from_masking_view(masking_view))
+                    # The storage group is the parent SG
+                    # Get the list of child SGs
+                    child_sg_list = (
+                        conn.provisioning.get_child_storage_groups_from_parent(
+                            mv_components['storagegroup']))
+                    element_dict, mv_components['storagegroup'] = (
+                        utils.choose_sg(masking_view, child_sg_list,
+                                        mv_components['portgroup'],
+                                        mv_components['initiatorgroup'],
+                                        is_revert))
+                    # Check if masking view exists and if it does validate it
+                    if element_dict:
+                        new_masking_view_details = (
+                            utils.get_or_create_masking_view(
+                                element_dict, mv_components['portgroup'],
+                                mv_components['initiatorgroup'], is_revert))
+                    else:
+                        utils.smart_print(
+                            'NO MIGRATION', migrate_utils.WARNING)
+                        sys.exit()
+
+            else:
+                if utils.check_mv_for_migration(masking_view):
+                    no_action = False
+                    utils.smart_print(
+                        'OLD MASKING VIEW IS %s',
+                        migrate_utils.DEBUG, masking_view)
+                    mv_components = (
+                        utils.get_elements_from_masking_view(masking_view))
+                    # Compile the new names of the SGs and MV
+                    element_dict = utils.compile_new_element_names(
+                        masking_view, mv_components['portgroup'],
+                        mv_components['initiatorgroup'],
+                        mv_components['storagegroup'])
+                    # Check if masking view exists and if it does validate it
+                    new_masking_view_details = (
+                        utils.get_or_create_masking_view(
+                            element_dict, mv_components['portgroup'],
+                            mv_components['initiatorgroup']))
+            # Get the volumes in the storage group
+            if 'storagegroup' in mv_components:
```

Comment:
mv_components may be referenced before assignment, its in the first if block but the else block has an if in there also that may not be met